web/src/features/charts/tooltips/CarbonChartTooltip.tsx

```diff
@@ -4,28 +4,31 @@ import { useCo2ColorScale } from 'hooks/theme';
 import { useAtom } from 'jotai';
 import { useTranslation } from 'react-i18next';
 import { formatDate } from 'utils/formatting';
-import { timeAverageAtom } from 'utils/state/atoms';
+import { productionConsumptionAtom, timeAverageAtom } from 'utils/state/atoms';
 import { InnerAreaGraphTooltipProps } from '../types';
+import { Mode } from 'utils/constants';
 
 export default function CarbonChartTooltip(props: InnerAreaGraphTooltipProps) {
   const [timeAverage] = useAtom(timeAverageAtom);
   const { i18n } = useTranslation();
   const { zoneDetail } = props;
+  const [currentMode] = useAtom(productionConsumptionAtom);
+  const isConsumption = currentMode === Mode.CONSUMPTION;
 
   const co2ColorScale = useCo2ColorScale();
 
   if (!zoneDetail) {
     return null;
   }
-  const { co2intensity, stateDatetime } = zoneDetail;
-
+  const { co2intensity, co2intensityProduction, stateDatetime } = zoneDetail;
+  const intensity = isConsumption ? co2intensity ?? 0 : co2intensityProduction ?? 0;
```

Comment:
_Nitpick alert:_
Could it work like this? Doesn't change much, but maybe a bit easier to read :)
```suggestion
  const intensity = isConsumption ? co2intensity ?? 0 : co2intensityProduction ?? 0;
  const intensity = (isConsumption ? co2intensity: co2intensityProduction) ?? 0;
```

Reply:
I was trying to one line this, I'll test it and see if it works